File: 3461. Check If Digits Are Equal in String After Operations I.cpp
class Solution {
    public:
        bool hasSameDigits(string s) 
        {
            int n = s.size();
            while(n > 2)
            {
                string temp;
                for(int i = 1; i < n; i++)
                    temp += to_string(((s[i] - '0') + (s[i - 1] - '0')) % 10);
                s = temp;
                n = s.size();
            }
            return s[0] == s[1];
        }
    };

File: 3461. Check If Digits Are Equal in String After Operations I.py
class Solution:
    def hasSameDigits(self, s: str) -> bool:
        if len(s) == 2:
            return s[0] == s[1]
        
        new_s = ""
        for i in range(1, len(s)):
            num = (int(s[i]) + int(s[i - 1])) % 10
            new_s += str(num)
        return self.hasSameDigits(new_s)